### ISSUE TITLE
Add options to set ulimit configures

### ISSUE THREAD

**Issue by robsonpeixoto** [poise/poise-service#42]
Some process need to open to many files, like prometheus, and you need to configure the `number of files` a process can open.
But the poise-service to have this feature.

Some useful links:
* upstart: http://bazaar.launchpad.net/~upstart-devel/upstart/trunk/view/head:/init/man/init.5#L906
* systemd: https://www.freedesktop.org/software/systemd/man/systemd.exec.html#LimitCPU=

The `sysvinit` and `inittab` just need to execute the `ulimit` command on the script.

But what's the better way to implement it?  As a `provider` option or as a `poise_service` option?

Thanks

**Comment by coderanger** [comment]
Is there a reason to not do this in the command for the service?

You could use something like `command "bash -c 'ulimit -whatever && myapp'"` or something like `chpst` from Runit. If you really want to do it in service framework, I would make a custom service template and point at that.

**Comment by robsonpeixoto** [comment]
@coderanger It'll work, but IMO is not a elegant solution.

And is better add a `exec` before execute myapp. It'll avoid a lot of errors with signal `command "bash -c 'ulimit -whatever && exec myapp'"`

**Comment by coderanger** [comment]
Yeah, something like that :) Not against supporting it as a provider option per se but I'm trying to avoid tooooo much template sprawl.

**Comment by robsonpeixoto** [comment]
If you would like, I can implement it as a *provider option* and open a PR.

**Comment by coderanger** [comment]
That would be a long term project to figure out how best to integrate so I would just use one of the workarounds from above for now.

**Comment by robsonpeixoto** [comment]
I'm already using the `bash -c ...` workarround.

**Comment by alicebob** [comment]
But `ulimit && mything` won't play nice with a non-root `user=` argument. (I need to set the number of open files). Even having a free-form key-value hash which ends up in the config would already be nice.

**Comment by coderanger** [comment]
@alicebob Indeed, you can use a custom template file for all the core providers to fully override things not otherwise exposed in the API.
